### ISSUE TITLE
OPC-UA Listener plugin support for Events

### ISSUE THREAD

**Issue by DaddyBearJon** [influxdata/telegraf#12518]
Would be nice to be able to subscribe to events (alarms) as well as data values. 
Generally speaking, you subscribe to OPC UA Events similarly to data changes, but by using monitoring parameters that contain an event filter instead of data change filter. Besides the DataChangeFilter property, the UAMonitoringParameters class also has an EventFilter property (of UAEventFilter type).

**Comment by srebhan** [comment]
@LarsStegman I think you contributed the OPC-UA listener. What do you think about the above proposal?

**Comment by LarsStegman** [comment]
Yes, that's definitely useful. We have something similar (far down) our wish list as well, so we would use it as well if it was implemented. 

I am not sure how much work it would be to implement, but if the gopcua library has support for it, I suspect it will not be too much work.

**Comment by srebhan** [comment]
I'm not deep into OPC-UA, but [this example](https://github.com/gopcua/opcua/pull/349/files) looks like it is supported by the gopcua lib... Volunteers for implementing this are welcome!

**Comment by DaddyBearJon** [comment]
I'm a control engineer and know OPC UA fairly well, I've only just started playing with Go, so might not be the best for implementing this!

 [This example](https://github.com/gopcua/opcua/blob/main/examples/subscribe/subscribe.go) has a working model, if you change a few parameters. I'm currently playing with it to pull alarms and events from our Siemens PCS7 DCS. Got it working on a Windows client but not on a Linux one. (Strange). I'll raise that on the Gopcua page.

I think the biggest issue will be deciding how to apply a suitable filter for the event data that comes through. (Each manufacturer can provide their own elements).

**Comment by srebhan** [comment]
@DaddyBearJon if you can get working code, I can walk you through creating a PR, improvements etc...

**Comment by DaddyBearJon** [comment]
@srebhan, I'll have a go :-). Really busy at work at the moment, we have a major overhaul outage every three years, which is coming up in April and most of my time is diverted to that at the moment. But when things ease up I will certainly take a look.

**Comment by srebhan** [comment]
No worries. Take your time. Let me know once you have something going...

**Comment by cesuper** [comment]
> Hey! I would like to read more about this topic. Any updates on event support using telegraf?

**Comment by JavaRookie** [comment]
Hey! I would like to read more about this topic. Any updates on event support using telegraf?
